### ISSUE TITLE
Exit non-zero on `tsh status` for scripting login status.

### ISSUE THREAD

**Issue by rduplain** [gravitational/teleport#5558]
### What

Exit non-zero on `tsh status` when no valid active profile exists: not logged in, no active profile, active profile expired. Optionally require a command-line flag for "query mode."

### How

```patch
--- a/tool/tsh/tsh.go
+++ b/tool/tsh/tsh.go
@@ -1794,15 +1794,27 @@ func printStatus(debug bool, p *client.ProfileStatus, isActive bool) {
 func onStatus(cf *CLIConf) error {
 	// Get the status of the active profile as well as the status
 	// of any other proxies the user is logged into.
+	//
+	// Return error if not logged in, no active profile, or expired.
 	profile, profiles, err := client.Status("", cf.Proxy)
 	if err != nil {
 		if trace.IsNotFound(err) {
 			fmt.Printf("Not logged in.\n")
-			return nil
 		}
 		return trace.Wrap(err)
 	}
+
 	printProfiles(cf.Debug, profile, profiles)
+
+	if profile == nil {
+		return trace.NotFound("no active profile")
+	}
+
+	duration := time.Until(profile.ValidUntil)
+	if duration.Nanoseconds() <= 0 {
+		return trace.NotFound("active profile expired")
+	}
+
 	return nil
 }
```

Developed against e235dfa35a1067a0475a28c5a0bcc0a1de776c48.

(The error messaging needs review, as this patch retains "Not logged in." in order to maintain the workaround below, while the NotFound error results in a redundant message "error: not logged in".)

### Why

Query login status before attempting login.

### Workaround

The most direct method currently is to just call `tsh login ...`, which will print status if already logged in and begin the login process if not.

To query only, assuming only one profile:

* `tsh status` output of "valid for" means logged in.
* `tsh status` output of "Not logged in" or "EXPIRED" means not logged in, as does empty output.

```sh
if tsh status | grep -q "valid for"; then
    echo "Logged in."
else
    echo "Not logged in."
fi
```

**Comment by rduplain** [comment]
The primary question in converting this to a PR: which is preferred?

1. Run with the patch posted in the description of this issue and remove `fmt.Printf("Not logged in.\n")`.
2. Leave `tsh status` as-is add a separate command or CLI flag for this "query mode."

With 6.0 coming up, it's an opportunity change `tsh status` behavior and go with (1).

**Comment by webvictim** [comment]
I think we would err on the side of not adding more flags - it just makes `tsh` usage more complicated. Please feel free to submit a PR with your patch and we'll review it. Thanks!

**Comment by webvictim** [comment]
@rduplain I think we would err on the side of not adding more flags - it just makes `tsh` usage more complicated. Please feel free to submit a PR with your patch and we'll review it. Thanks!

**Comment by rduplain** [comment]
Merged in #6957, thanks!
